### DISCUSSION TITLE
How to use multipoint tutorial file for DARhoSimpleCFoam?

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#110] [Tutorials]
I am interested to perform multipoint optimization for subsonic, transonic and supersonic flight conditions at different altitudes. 
Multipoint tutorial use DASimpleFoam, which is incompressible. Kindly help me in this regard.

**Comment by friedenhe** [comment]
We currently do not support switching solvers during optimization. For your case, I suggest you try DARhoSimpleCFoam. It is mainly for transonic and (weak) supersonic cases, but it may also work for subsonic conditions. You need to use the DASimpleFoam tutorial as reference and change flight conditions and the solverName to DARhoSimpleCFoam. Again, I suggest you go over other single-point optimization tutorials before trying the multipoint one.

**Reply by usmanshahid90** [reply]
It's working by doing certain changes to code.  But I am confused that the "opt_SLSQP.out" file is giving values of drag coefficients for various designs. What are the flight conditions for these results? are they kind of some averaged? or something else?

**Reply by friedenhe** [reply]
Check this [page](https://dafoam.github.io/mydoc_tutorials_aero_naca0012_multipoint.html)

**Reply by usmanshahid90** [reply]
Yea, I have visited that page.. it says " Combined Drag".. Whats the meaning of combined drag? Is it something like weighted average?

**Reply by friedenhe** [reply]
Yes, it is a weighted average.

**Reply by usmanshahid90** [reply]
Great.. Thanks

**Reply by usmanshahid90** [reply]
One more thing.

Can I get Cd results for all the flight conditions separately along with weighted average? Or i need to do single point optimizations for all those flight conditions one by one to get those results?

**Reply by friedenhe** [reply]
Yes, the CD and CL for all flight conditions will be printed on the screen. Read [this](https://dafoam.github.io/mydoc_get_started_runscript.html#optimization-progress-logopttxt)

**Reply by wndzjl** [reply]
I tried using `setOption("solverName", "DASimpleFoam")` to change the solver between different points, but it didn’t work. I have an idea, and I would appreciate Professor He’s feedback on whether it’s possible to set both `DASolver = PYDAFOAM(options=daOptions, comm=MPI.COMM_WORLD)` and `DASolver1 = PYDAFOAM(options=daOptions1, comm=MPI.COMM_WORLD)` simultaneously. They would share the same `DVCon` and `DVGeo`, and then modify the `calcObjFuncValuesMP( )` and `calcObjFuncSensMP( )` in `optFuncs.py` so that within the `for i in range(nMultiPoints) loop`, different solvers are executed. This is the general idea I have, and I would appreciate Professor He’s assessment on whether any modifications are needed.

**Reply by friedenhe** [reply]
@wndzjl If you want to use different solvers in an optimization, you should take a look at the NACA0012/multi_case tutorial.

**Reply by wndzjl** [reply]
Does this work with version v2 code? That is, do not use `openmdao`

**Reply by friedenhe** [reply]
Nope. It works for v3+ only

**Reply by wndzjl** [reply]
Sorry, I meant if it can be used with code types like` runScript_v2.py` instead of `runScript.py`. That is, do not use `openmdao.api, mphys.multipoint, dafoam.mphys, mphys.scenario_aerodynamic` libraries, etc. Because this version is a little hard to understand.

**Reply by friedenhe** [reply]
Nope, the multicase is available for v3 scripts only
